### PR TITLE
DOC: Add an example showing how to plot Rotations to the docs

### PR DESCRIPTION
**Pull request by scottshambaugh**
#### What does this implement/fix?
This adds an example of how to plot Rotations to the main Rotation object docstring. This also serves as an example of intrinsic vs extrinsic Euler angle conventions.

### CODE REVIEW

**Comment by scottshambaugh** [comment]
Anyone have advice on getting larger images to render without cluttering up the docstrings? Otherwise this is good to go.

Image: 
image

Code:
```python
from scipy.spatial.transform import Rotation as R
import numpy as np
import matplotlib.pyplot as plt

def plot_rotated_axes(ax, R, name=None, offset=(0, 0, 0), scale=1):
    loc = np.array([offset, offset])
    for i, (axis, c) in enumerate(zip((ax.xaxis, ax.yaxis, ax.zaxis),
                                      ("r", "g", "b"))):
        axlabel = axis.axis_name
        axis.set_label_text(axlabel)
        axis.label.set_color(c)
        axis.set_tick_params(colors=c)
        line = np.zeros((2, 3))
        line[1, i] = scale
        line_rot = R.apply(line)
        line_plot = line_rot + loc
        ax.plot(line_plot[:, 0], line_plot[:, 1], line_plot[:, 2], c)
        text_loc = line[1]*1.2
        text_loc_rot = R.apply(text_loc)
        text_plot = text_loc_rot + loc[0]
        ax.text(*text_plot, axlabel, color=c, va="center", ha="center")
    ax.text(*offset, name, color="k", va="center", ha="center",
            bbox={"fc": "w", "alpha": 0.8, "boxstyle": "circle"})

r0 = R.identity()
r1 = R.from_euler("ZYX", [90, -30, 0], degrees=True)  # intrinsic
r2 = R.from_euler("zyx", [90, -30, 0], degrees=True)  # extrinsic

ax = plt.figure().add_subplot(projection="3d", proj_type="ortho")
plot_rotated_axes(ax, r0, name="r0", offset=(0, 0, 0))
plot_rotated_axes(ax, r1, name="r1", offset=(3, 0, 0))
plot_rotated_axes(ax, r2, name="r2", offset=(6, 0, 0))
ax.set(xlim=(-1.25, 7.25), ylim=(-1.25, 1.25), zlim=(-1.25, 1.25))
ax.set_aspect("equal", adjustable="box")
plt.tight_layout()

plt.show()
```

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for the PR @scottshambaugh. This is a good start. I would suggest:

- remove the axis ticks as I don't think it brings much (maybe even the grid, or make it sparser),
- change colours to be more colour blind friendly (RGB is not great for that matter),
- make the circles opaque to better see the names,
- add a text explaining in plain English what we are seeing.

**Comment by scottshambaugh** [comment]
Thanks @tupui, those are good suggestions and I'll clean it up. The only thing I would push back on is the coloring, since XYZ coordinate axes being colored RGB respectively is [a generally accepted standard convention](https://ux.stackexchange.com/questions/79561/why-are-x-y-and-z-axes-represented-by-red-green-and-blue). I think that the x, y, z, floating text at the end of each line suffices to resolve colorblindness ambiguities.

**Comment by tupui** [comment]
> The only thing I would push back on is the coloring, since XYZ coordinate axes being colored RGB respectively is [a generally accepted standard convention](https://ux.stackexchange.com/questions/79561/why-are-x-y-and-z-axes-represented-by-red-green-and-blue). I think that the x, y, z, floating text at the end of each line suffices to resolve colorblindness ambiguities.

Sorry no 😄 we are not compromising on accessibility even if there is a convention (we should actually promote better ones, but not the topic here OC). RGB is the worst we can do in terms of colour blindness as colour blindness specifically affect these 3 colours. See this nice post for instance https://www.datylon.com/blog/data-visualization-for-colorblind-readers and this website for some good colour palette options https://colorbrewer2.org

**Comment by scottshambaugh** [comment]
Heard, I'll play around with something that accommodates everything, flagging [this link](https://davidmathlogic.com/colorblind/#%233821AD-%23117752-%23FF523C) to look at when I get home later.

**Review by tupui — COMMENTED** [review]
@scottshambaugh thanks for updating the PR.

Please add the following in your commit messages `[skip cirrus] [skip actions] [skip azp]`. Also don't force push, when merging we will squash merge. This is important for the review process.

You can find these info and more in our contributing guide 😃 https://scipy.github.io/devdocs/dev/contributor/contributor_toc.html

**Comment by scottshambaugh** [comment]
Hi @tupui, made some updates to address your comments. Thanks for the tags to skip parts of CI, though I've read the contributors guide and looking and I can't find any guidance on not force pushing on feature branches. Not a problem, though I don't know of another way to kick off CI again without making any changes. Here's the new plot as rendered in the docs:

scipy-spatial-transform-Rotation-1_00_00

Some notes:
1) You need some transparency in the circles or they complete obscure the axes when those are coming towards the camera.
2) I believe the ticks are important as they establish positive directionality to the base axes which is necessary for proper orientation, however I reduced them to the whole numbers to reduce clutter.
3) I landed on [this color palette](https://davidmathlogic.com/colorblind/#%23FF6666-%23005533-%231199EE) for the axes, which I think should be sufficiently colorblind-safe. This color set also varies in lightness for black & white printing, which was hard to find in other colorblind-safe sets. I also checked with [this tool](https://www.color-blindness.com/coblis-color-blindness-simulator/) that the final image had distinguishable coloring under the all different forms of colorblindness. I am very reticent to completely move away from something that parses as "RGB" for the 95% of people with full color vision, since there is semantic meaning for coordinate axes specifically, so hopefully this suffices.
image

```python
from scipy.spatial.transform import Rotation as R
import numpy as np
import matplotlib.pyplot as plt

def plot_rotated_axes(ax, r, name=None, offset=(0, 0, 0), scale=1):
    colors = ("#FF6666", "#005533", "#1199EE")  # Colorblind-safe RGB
    loc = np.array([offset, offset])
    for i, (axis, c) in enumerate(zip((ax.xaxis, ax.yaxis, ax.zaxis),
                                      colors)):
        axlabel = axis.axis_name
        axis.set_label_text(axlabel)
        axis.label.set_color(c)
        axis.set_tick_params(colors=c)
        axis.line.set_color(c)

        line = np.zeros((2, 3))
        line[1, i] = scale
        line_rot = r.apply(line)
        line_plot = line_rot + loc
        ax.plot(line_plot[:, 0], line_plot[:, 1], line_plot[:, 2], c)

        text_loc = line[1]*1.2
        text_loc_rot = r.apply(text_loc)
        text_plot = text_loc_rot + loc[0]
        ax.text(*text_plot, axlabel.upper(), color=c,
                va="center", ha="center")

    ax.text(*offset, name, color="k", va="center", ha="center",
            bbox={"fc": "w", "alpha": 0.8, "boxstyle": "circle"})

r0 = R.identity()
r1 = R.from_euler("ZYX", [90, -30, 0], degrees=True)  # intrinsic
r2 = R.from_euler("zyx", [90, -30, 0], degrees=True)  # extrinsic

ax = plt.figure().add_subplot(projection="3d", proj_type="ortho")
plot_rotated_axes(ax, r0, name="r0", offset=(0, 0, 0))
plot_rotated_axes(ax, r1, name="r1", offset=(3, 0, 0))
plot_rotated_axes(ax, r2, name="r2", offset=(6, 0, 0))

ax.annotate("r0: Identity Rotation\n"
            "r1: Intrinsic Euler Rotation (ZYX)\n"
            "r2: Extrinsic Euler Rotation (zyx)",
            xy=(0.6, 0.7), xycoords="axes fraction", ha="left")

ax.set(xlim=(-1.25, 7.25), ylim=(-1.25, 1.25), zlim=(-1.25, 1.25))
ax.set(xticks=range(-1, 8), yticks=[-1, 0, 1], zticks=[-1, 0, 1])
ax.set_aspect("equal", adjustable="box")
ax.figure.set_size_inches(6, 5)
plt.tight_layout()

plt.show()
```

**Comment by tupui** [comment]
The normal workflow is just to add commits to the branch and push the new commits. This would trigger again the CI.

Don't worry too much about getting the whole CI green. On large projects like SciPy, a lot of things are breaking all the time 😅 Just read the log and see if that could be related to your changes or not. And if you are not sure you can always ask 😃 With time/exp. you will know what to look for, etc.

`Main prerelease_deps_coverage_64bit_blas` is a known issue at the moment. It's due to a pre-releases of Cython. Nothing to worry about here.

For doc only PRs like this one. Typically you just want to run CircleCI (builds the doc) and from time to time Azure CI (linters and doctests). Meaning you would either do `[skip cirrus] [skip actions] [skip azp]` or `[skip cirrus] [skip actions]`.

---

Thank you for the update! This looks great now 🚀 

Ok for the transparency, colours look good. I could nitpick on the x tick touching the numbers but it's really fine as readable. I will check the rendered doc and do a better review next week as I am done for today.

File: scipy/spatial/transform/_rotation.pyx [hunk 1]
```diff
@@ -617,6 +617,61 @@ cdef class Rotation:
...
+    Add all three rotations to a single plot:
+
+    >>> ax = plt.figure().add_subplot(projection="3d", proj_type="ortho")
+    >>> plot_rotated_axes(ax, r0, name="r0", offset=(0, 0, 0))
+    >>> plot_rotated_axes(ax, r1, name="r1", offset=(3, 0, 0))
+    >>> plot_rotated_axes(ax, r2, name="r2", offset=(6, 0, 0))
+    >>> ax.annotate("r0: Identity Rotation\\n"
+    ...             "r1: Intrinsic Euler Rotation (ZYX)\\n"
+    ...             "r2: Extrinsic Euler Rotation (zyx)",
+    ...             xy=(0.6, 0.7), xycoords="axes fraction", ha="left")
```
Comment (tupui) [inline scipy/spatial/transform/_rotation.pyx:664]: Fixing the refguide check
```suggestion
    >>> _ = ax.annotate(
    ...     "r0: Identity Rotation\\n"
    ...     "r1: Intrinsic Euler Rotation (ZYX)\\n"
    ...     "r2: Extrinsic Euler Rotation (zyx)",
    ...     xy=(0.6, 0.7), xycoords="axes fraction", ha="left"
    ... )
```

**Review by tupui — APPROVED** [review]
Thank you for the update @scottshambaugh. If the CI is back green I would merge. We can iterate further if needed but this is looking good to me already.
